src/field255.rs

```diff
@@ -0,0 +1,305 @@
+//! Finite field arithmetic for `GF(2^255 - 19)`.
```

Comment:
I was hoping there would be a higher-level API for arithmetic in this field that we could take advantage of, i.e., that we wouldn't use fiat-crypto directly. However after doing some research, this appears not to be the case.

The best rust implementation of GF(2^255-19) is dalek, but they don't expose the field arithemtic directly. I also noticed that the code here is essentially a re-implementation of [dalek's old fiat backend](https://github.com/dalek-cryptography/curve25519-dalek/blob/main/src/backend/serial/fiat_u64/field.rs). As I understand it, maintenance has moved to the zkcrypto project (https://github.com/zkcrypto/curve25519-dalek-ng). Interestingly, there the maintainers have removed the fiat backend entirely.

I'm not sure what the best path forward here would be. If the zkcrypto folks are willing to consider exposing the field arithmetic (https://github.com/zkcrypto/curve25519-dalek-ng/issues/20), then I think that would be our best bet. In the meantime, re-implementing the fiat backend is probably our best option.

This code is delicate and will require careful review. Let's move splitting the FFT functionality of `FieldElement` and implementing `Field255` into a separate PR, then rebase this PR on top. I will see if I can get some feedback from folks on my team who have experience with fiat-crypto.